File: Demo/TyphoonRestClientDemo/TyphoonRestClientDemo/ObjectMappers/TRCObjectMapperIssue.m


#import "TRCObjectMapperIssue.h"
#import "Issue.h"

@implementation TRCObjectMapperIssue

- (id)objectFromResponseObject:(NSDictionary *)dict error:(NSError **)error;
{
    Issue *issue = [Issue new];
    issue.identifier = dict[@"id"];
    issue.projectName = dict[@"project"][@"name"];
    issue.authorName = dict[@"author"][@"name"];
    issue.statusText = dict[@"status"][@"name"];
    issue.subject = dict[@"subject"];
    issue.descriptionText = dict[@"description"];
    issue.updated = dict[@"created_on"];
    issue.created = dict[@"updated_on"];
    issue.doneRatio = dict[@"done_ratio"];
    return issue; //You allowed to return nil. If items in the array, this item would be skipped
}

@end

File: Demo/TyphoonRestClientDemoSwift/TyphoonRestClientDemoSwift/TyphoonRestClientDemoSwift/ObjectMappers/TRCObjectMapperIssue.swift
////////////////////////////////////////////////////////////////////////////////
//
//  APPS QUICKLY
//  Copyright 2015 Apps Quickly Pty Ltd
//  All Rights Reserved.
//
//  NOTICE: Prepared by AppsQuick.ly on behalf of Apps Quickly. This software
//  is proprietary information. Unauthorized use is prohibited.
//
////////////////////////////////////////////////////////////////////////////////


import Foundation

class TRCObjectMapperIssue: NSObject, TRCObjectMapper {
    
    func objectFromResponseObject(responseObject: AnyObject!, error: NSErrorPointer) -> AnyObject! {
        
        let responseDictionary = responseObject as! NSDictionary
        
        var issue = Issue()
        issue.identifier = responseDictionary["id"] as! Int
        issue.projectName = responseDictionary.valueForKeyPath("project.name") as? String
        issue.authorName = responseDictionary.valueForKeyPath("author.name") as? String
        issue.statusText = responseDictionary.valueForKeyPath("status.name") as? String
        issue.subject = responseDictionary["subject"] as! String
        issue.descriptionText = responseDictionary["description"] as! String
        issue.updated = responseDictionary["created_on"] as? NSDate
        issue.created = responseDictionary["updated_on"] as? NSDate
        issue.doneRatio = responseDictionary["done_ratio"] as! Double
        
        return issue
        
    }
}